include/cudaq/Optimizer/Transforms/ArgumentDataStore.h

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include <cassert>
+#include <numbers>
+#include <tuple>
+#include <vector>
+
+namespace cudaq::opt {
+
+/// Stores argument values.
+/// Stores data, element size, and number of elements for each argument.
+/// If the data pointer is owned, cleans up the data on destruction.
```

Comment:
Is this here as some sort of wrapper around all argument pointers? I'm not sure why it is needed though. The only argument type that seems relevant to this PR is a `cudaq::state *`.

This seems fully redundant with the new argument synthesis approach, unfortunately.

Reply:
Yes this is just to store state data from the state* arguments. Let me know if you would like this to do in before the new approach is finished - I can definitely remove it later when not needed anymore.